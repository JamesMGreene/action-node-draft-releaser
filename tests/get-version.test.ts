import path from 'path';
import getVersion from '../src/get-version';

describe('getVersion', () => {
  afterEach(() => {
    delete process.env.GITHUB_WORKSPACE;
  });

  it('should return the version from the "package.json" in the workspace', async () => {
    process.env.GITHUB_WORKSPACE = path.join(__dirname, 'fixtures');

    const version = await getVersion();
    expect(version).toBe('1.0.0-beta.2');
  });

  it('should throw if no "package.json" is found in the workspace', () => {
    process.env.GITHUB_WORKSPACE = path.join(__dirname, 'nonexistent-fixtures');

    expect(() => getVersion()).toThrowErrorMatchingSnapshot();
  });
});
