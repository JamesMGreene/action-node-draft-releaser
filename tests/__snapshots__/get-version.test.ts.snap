// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getVersion should throw if no "package.json" is found in the workspace 1`] = `"\\"package.json\\" could not be found in your project's root."`;
